src/store/modules/subscriptionStore.js

```diff
@@ -64,20 +74,56 @@ export const subscriptionStore = {
 
             const url = process.env.BASE_URL + "report-recipe";
 
-            this.state.api.client.post(url, {
-                payload
-            }, this.state.api.options, {  headers: {
-                'Authorization': `Bearer ${this.state.access_token}`
-            }})
-            .then((response) => {
-                context.commit('SET_MODAL_STATE', false)
-                context.commit('SHOW_REPORT_SUCCESS_MESSAGE')
-            })
-            .catch((error) => {
-                context.commit('SET_MODAL_STATE', false)
-                context.commit('SHOW_FAILED_MESSAGE')
-                context.commit("HANDLE_ERROR", error.response);
-            })
+            this.state.api.client
+                .post(
+                    url,
+                    {
+                        payload,
+                    },
+                    this.state.api.options,
+                    {
+                        headers: {
+                            Authorization: `Bearer ${this.state.access_token}`,
+                        },
+                    },
+                )
+                .then((response) => {
+                    context.commit("SET_MODAL_STATE", false);
+                    context.commit("SHOW_REPORT_SUCCESS_MESSAGE");
+                })
+                .catch((error) => {
+                    context.commit("SET_MODAL_STATE", false);
+                    context.commit("SHOW_FAILED_MESSAGE");
+                    context.commit("HANDLE_ERROR", error.response);
+                });
+        },
+        submit_report_otp(context, payload) {
+            context.commit("SET_MODAL_STATE", true);
+
+            const url = process.env.BASE_URL + "otp"; //not actual endpoint
+
+            this.state.api.client
+                .post(
+                    url,
+                    {
+                        payload,
+                    },
+                    this.state.api.options,
```

Comment:
Is ther eany need for this line? That and the block after it are doing the same exact thing. Remove the api.options one let's set the authorization header manually